benbarnett02.CodeTrack/TrackDB/TrackingData.cs

```diff
@@ -0,0 +1,211 @@
+﻿using Microsoft.Data.Sqlite;
+using System.Globalization;
+using System.Configuration;
+using System.Collections.Generic;
+using static TrackingProgram.TrackingData;
+
+
+
+// This class handles all database operations. 
+namespace TrackingProgram;
+
+public class TrackingData
+{
+    public static string connectionString = ConfigurationManager.ConnectionStrings["TrackDB"].ConnectionString;
+    public static void StartupDatabase()
+    {
+        using (var trackingDatabase = new SqliteConnection(connectionString))
+        {
+            trackingDatabase.Open();
+            var tableCmd = trackingDatabase.CreateCommand();
+
+            tableCmd.CommandText =
+                @"CREATE TABLE IF NOT EXISTS coding_entries (
+Id INTEGER PRIMARY KEY AUTOINCREMENT,
+Label TEXT,
+StartDate TEXT,
+EndDate TEXT
+)";
+
+            tableCmd.ExecuteNonQuery();
+            trackingDatabase.Close();
+
+        }
+    }
+
+    public static List<CodeEntry> GetAllCodeRecords()
+    {
+        using (var myDatabase = new SqliteConnection(connectionString))
+        {
+            myDatabase.Open();
+            var tableCmd = myDatabase.CreateCommand();
+
+            tableCmd.CommandText =
+                $"SELECT * FROM coding_entries";
+
+            List<CodeEntry> tableData = new();
+            SqliteDataReader reader = tableCmd.ExecuteReader();
+            if (reader.HasRows)
+            {
+                while (reader.Read())
+                {
+                    tableData.Add(
+                        new CodeEntry
+                        {
+                            Id = reader.GetInt32(0),
+                            Label = reader.GetString(1),
+                            StartDate = DateTime.ParseExact(reader.GetString(2), "dd/MM/yyyy HH:mm tt", new CultureInfo("en-US")),
+                            EndDate = DateTime.ParseExact(reader.GetString(3), "dd/MM/yyyy HH:mm tt", new CultureInfo("en-US")),
+                        }
+                        );
+                }
+                myDatabase.Close();
+                return tableData;
+            }
+            else
+            {
+                myDatabase.Close();
+                return null;
+            }
+        }
+    }
+
+    public static void InsertCodeEntry(CodeEntry codeEntry)
+    {
+        string format = "dd/MM/yyyy HH:mm tt";
+        using (var myDatabase = new SqliteConnection(connectionString))
```

Comment:
![image](https://user-images.githubusercontent.com/108711476/207208156-88558aec-4a77-43f3-bc07-e79615f162b6.png)

There needs to be a check to make sure the end date isn't before the start date, which an error message accordingly.